Utilities/CreateServicePrincipal.ps1

```diff
@@ -223,16 +223,34 @@ function New-ServicePrincipal() {
     Write-Prompt "2) Owner"
     Write-Prompt "Please choose by entering 1 or 2:"
     $Privilege = Read-Host
-
     Write-Host "Assign roles to Service Principal..."
-    if ($Privilege -eq 2) {
-        New-AzRoleAssignment -RoleDefinitionName "Owner" -ApplicationId $ClientId
-    }
-    else {
-        New-AzRoleAssignment -RoleDefinitionName "Contributor" -ApplicationId $ClientId
+    try {
+        if ($Privilege -eq 2) {
+            New-AzRoleAssignment -RoleDefinitionName "Owner" -ApplicationId $ClientId
+        }
+        else {
+            New-AzRoleAssignment -RoleDefinitionName "Contributor" -ApplicationId $ClientId
+        }
+        Write-Host "Successfully created Service Principal and assigned role...`n"
+    } catch {
+        $ErrorMessage = $_.Exception.Message
+        Write-Host -ForegroundColor Red "$ErrorMessage"
+        $currentSignInAccount = Get-AzContext | Select-Object Account
+        $role = Get-AzRoleAssignment -SignInName $currentSignInAccount.Account.id | Select-Object RoleDefinitionName
+        # https://docs.microsoft.com/en-us/powershell/azure/create-azure-service-principal-azureps?view=azps-2.5.0#manage-service-principal-roles
+        if ($ErrorMessage -match "does not have authorization to perform action 'Microsoft.Authorization/roleAssignments/write' over scope") {
```

Comment:
Using 'message' text string in the conditional statement is not recommended. Can you switch to the error code?

Reply:
I use $_